src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -686,9 +686,9 @@ private void ExecuteOperationAction(
         /// Execute code block actions for the given analyzer for the given declaration.
         /// </summary>
         public void ExecuteCodeBlockActions<TLanguageKindEnum>(
-            IEnumerable<CodeBlockStartAnalyzerAction<TLanguageKindEnum>> codeBlockStartActions,
-            IEnumerable<CodeBlockAnalyzerAction> codeBlockActions,
-            IEnumerable<CodeBlockAnalyzerAction> codeBlockEndActions,
+            ImmutableArray<CodeBlockStartAnalyzerAction<TLanguageKindEnum>> startActions,
+            ImmutableArray<CodeBlockAnalyzerAction> actions,
+            ImmutableArray<CodeBlockAnalyzerAction> endActions,
```

Comment:
strongly typed, to prevent boxing and ensure fast enumeration. naming made consistent between all methods.